lasagne/tests/test_objectives.py

```diff
@@ -1,65 +1,228 @@
+        assert abs(result_with_mask.eval() - 5.0/3.0) < 1.0e-9
+
+        # Masked NLL sum is 2 + 3 = 5
+        result_with_mask = self.get_masked_loss(categorical_crossentropy,
+                                                output, target_2d,
+                                                mask_1hot, aggregation='normalized_sum')
+        assert result_with_mask.eval() == 5.0/2.0
+
+
+    def test_objective(self):
+        from lasagne.objectives import Objective
+
+        input_layer = mock.Mock()
+        loss_function = mock.Mock()
+        input, target, arg1, kwarg1 = (object(),) * 4
+        objective = Objective(input_layer, loss_function)
+        result = objective.get_loss(input, target, 'mean', arg1, kwarg1=kwarg1)
+
+        # We expect that the input layer's `get_output` was called with
+        # the `input` argument we provided, plus the extra positional and
+        # keyword arguments.
+        input_layer.get_output.assert_called_with(input, arg1, kwarg1=kwarg1)
+        network_output = input_layer.get_output.return_value
+
+        # The `network_output` and `target` are fed into the loss
+        # function:
+        loss_function.assert_called_with(network_output, target)
+        assert result == loss_function.return_value.mean.return_value
+
+
+    def test_objective_no_target(self):
+        from lasagne.objectives import Objective
+
+        input_layer = mock.Mock()
+        loss_function = mock.Mock()
+        input = object()
+        objective = Objective(input_layer, loss_function)
+        result = objective.get_loss(input)
+
+        input_layer.get_output.assert_called_with(input)
+        network_output = input_layer.get_output.return_value
+
+        loss_function.assert_called_with(network_output, objective.target_var)
+        assert result == loss_function.return_value.mean.return_value
+
+
+    # def test_masked_objective(self):
+    #     from lasagne.objectives import MaskedObjective
+    #
+    #     input_layer = mock.Mock()
+    #     loss_function = mock.Mock()
+    #     mask = mock.Mock()
+    #     input, target, arg1, kwarg1 = (object(),) * 4
+    #     objective = MaskedObjective(input_layer, loss_function)
+    #     result = objective.get_loss(input, target, mask, False, arg1, kwarg1=kwarg1)
+    #
+    #     # We expect that the input layer's `get_output` was called with
+    #     # the `input` argument we provided, plus the extra positional and
+    #     # keyword arguments.
+    #     input_layer.get_output.assert_called_with(input, arg1, kwarg1=kwarg1)
+    #     network_output = input_layer.get_output.return_value
+    #
+    #     # The `network_output` and `target` are fed into the loss
+    #     # function:
+    #     loss_function.assert_called_with(network_output, target, mask)
+    #     loss_function.return_value.__mul__.assert_called_with(mask)
+    #     assert result == loss_function.return_value.__mul__.return_value.sum.return_value
```

Comment:
I'd prefer this to be removed, we don't want to start collecting dead code as Theano does :)